### PR TITLE
[file] new verbs 'isabsolute', 'normalize', 'cwd', 'patchpath'

### PR DESCRIPTION
**Pull request by umlaeute**
this adds 4 new subcommands to `[file]`:

- `[file normalize]` - normalize a path by removing redundant components
- `[file isabsolute]` - check whether a path is absolute 
   - Closes: #1535
- `[file cwd]` - get/set the current working directory
  - Closes: #1537
- `[file patchpath]` - get the patch's path (kind of obsoletes the `[dir(` message to `[pdcontrol]`) 
   - Closes: #1539

the `file-help.pd` reference has also been updated accordingly, which will most likely lead to conflicts with #1517.
i have first synched the reference patch with #1517 (as of 675591aacd684ad3f07dece2f7385a19d8624646), so i guess the resolution would be to use the `file-help.pd` of **this** PR.

### CODE REVIEW

**Comment by umlaeute** [comment]
out of mere habit this PR is based on `develop`...

**Comment by Spacechild1** [comment]
> out of mere habit this PR is based on develop...

why not rebase onto master? ;-)

File: src/x_file.c [hunk 1]
```diff
@@ -87,6 +93,30 @@ static int sys_remove(const char *pathname) {
     u8_utf8toucs2(ucs2buf, MAXPDSTRING, pathname, MAXPDSTRING);
     return do_delete_ucs2(ucs2buf);
 }
+static char* sys_getcwd(char *buf, size_t size) {
+    int err=0;
+    size_t ucs2size = size;
+    uint16_t*res = 0, *ucs2buf = getbytes(ucs2size*sizeof(*ucs2buf));
```
Comment (Spacechild1) [inline src/x_file.c:99]: why not simply use `uint16_t src[MAX_PATH]`, like in the functions above?

Comment (umlaeute) [reply src/x_file.c]: good point.
the reason was file's `sys_*` closely wrap the POSIX functions (which has a `size`) parameter, and it appeared to me that this dynamic size ought to be passed thru as far as possible.

anyhow, i've now switched to a stack-based allocation.

**Comment by umlaeute** [comment]
@millerpuckette what do you think about this? is this something that you do not want in principle? or that ought to be implemented differently? that needs more thinking?

has the ship sailed for new features in Pd-0.53?

note that this PR is actually driven by some requests by others (notably @Spacechild1; but also a few others chimed in)

**Comment by Spacechild1** [comment]
@millerpuckette It would be really great if this could make it into Pd 0.54!

**Pull request merged**